stumpy/core.py

```diff
@@ -2254,15 +2349,114 @@ def rolling_isconstant(a, w):
     w : numpy.ndarray
         The rolling window size
 
+    T_subseq_isconstant : np.ndarray or function, default None
+        A boolean array that indicates whether a subsequence in `T` is constant
+        (True). Alternatively, a custom, user-defined function that returns a
+        boolean array that indicates whether a subsequence in `T` is constant
+        (True). The function must only take two arguments, `a`, a 1-D array,
+        and `w`, the window size, while additional arguments may be specified
+        by currying the user-defined function using `functools.partial`. When
+        None, this defaults to `_rolling_isconstant`.
+
     Returns
     -------
-    output : numpy.ndarray
-        Rolling window isconstant.
+    T_subseq_isconstant : numpy.ndarray
+        Rolling window isconstant
     """
-    axis = a.ndim - 1
-    return np.apply_along_axis(
-        lambda a_row, w: _rolling_isconstant(a_row, w), axis=axis, arr=a, w=w
-    )
+    if T_subseq_isconstant is None:
+        T_subseq_isconstant = _rolling_isconstant
+
+    isconstant_func = None
+    if callable(T_subseq_isconstant):
+        non_default_args = []
```

Comment:
I don't know if we need/want to do it now but it feels like this argument comparison code would be potentially useful for other `core.rolling_func` functions and maybe it would be nice to pull it out into it's own function:

```
def is_signature_compatible(func, required_args):
    if not isinstance(required_args, list):
        required_args = list(required_args)

    non_default_args = []
    for arg_name, arg in inspect.signature(T_subseq_isconstant).parameters.items():
        # inspect.signature(functools.partial(func)) returns all arguments
        # including the ones with default values. the following if block
        # is to find non-default arguments.
        if arg.default == inspect.Parameter.empty:
            non_default_args.append(arg_name)

        incomp_args = set(non_default_args).difference(set(required_args))

    return len(incomp_args) > 0
```

Reply:
Right! I think that would be a good idea!! I will add it to the `core.py`